library/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -132,80 +227,266 @@
             if module.params['job_wait']:
-                msg['changed'] = True
+                remote['msg'] = 'Succesfully completed the repository update job.'
+                result['changed'] = True
         else:
-            module.fail_json(msg='Failed to update firmware.', update_status=msg['update_status'])
-    return msg
+            result['msg'] = 'Failed to update firmware.'
+            module.fail_json(**result)
+    else:
+        result['msg'] = 'Failed to update firmware.'
+        module.fail_json(**result)
+
+    return result
 
 
 def main():
+
     module = AnsibleModule(
         argument_spec={
+            # iDRAC Credentials
             "idrac_ip": {"required": True, "type": 'str'},
             "idrac_user": {"required": True, "type": 'str'},
             "idrac_pwd": {"required": True, "type": 'str', "no_log": True},
             "idrac_port": {"required": False, "default": 443, "type": 'int'},
 
+            # Network File Share
             "share_name": {"required": True, "type": 'str'},
-            "share_user": {"required": False, "type": 'str'},
-            "share_pwd": {"required": False, "type": 'str', "no_log": True},
-            "share_mnt": {"required": True, "type": 'str'},
+            "share_user": {"required": False, "type": 'str', "default": None},
+            "share_pwd": {"required": False, "type": 'str', "default": None, "no_log": True},
+            "share_mnt": {"required": False, "type": 'path', "default": None},
 
-            "catalog_file_name": {"required": False, "type": 'str', "default": "Catalog.xml"},
-            "reboot": {"required": False, "type": 'bool', "default": False},
-            "job_wait": {"required": False, "type": 'bool', "default": True},
+            # Firmware update parameters
+            "catalog_file_name": {"required": False, "default": 'Catalog.xml', "type": 'str'},
+            "apply_update": {"required": False, "default": True, "type": 'bool'},
+            "reboot": {"required": False, "default": False, "type": 'bool'},
+            "job_wait": {"required": False, "default": True, "type": 'bool'},
+            "ignore_cert_warning": {"required": False, "default": True, "type": 'bool'}
         },
 
         supports_check_mode=False)
 
     try:
-        # Validate the catalog file
-        _validate_catalog_file(module.params['catalog_file_name'])
+        # validate module arguments
+        _validate_args(module)
+
         # Connect to iDRAC and update firmware
         with iDRACConnection(module.params) as idrac:
-            update_status = update_firmware(idrac, module)
-    except (ImportError, ValueError, RuntimeError) as e:
+            result = update_firmware(idrac, module)
+
+    except (ImportError, ValueError, TypeError, RuntimeError) as e:
         module.fail_json(msg=str(e))
 
-    module.exit_json(msg='Successfully updated the firmware.', update_status=update_status)
+    module.exit_json(**result)
```

Comment:
would recommend you to `msg='Successfully updated the firmware.', update_status=update_status ` to be very definite about the data were are sending back. if **result there are possibility that some random data can get into this variable length dict. Not sure this is a security best practice or not